src/mx_bluesky/beamlines/i04/test_plans/oav_automation.py

```diff
@@ -0,0 +1,66 @@
+import bluesky.plan_stubs as bps
+from bluesky.utils import MsgGenerator
+from dodal.common import inject
+from dodal.devices.attenuator.attenuator import BinaryFilterAttenuator
+from dodal.devices.oav.oav_detector import OAV
+from dodal.devices.scintillator import InOut, Scintillator
+from dodal.devices.xbpm_feedback import XBPMFeedback
+from dodal.devices.zebra.zebra_controlled_shutter import (
+    ZebraShutter,
+    ZebraShutterControl,
+    ZebraShutterState,
+)
+
+# from dodal.devices.oav.snapshots.snapshot_with_grid import SnapshotWithGrid
+
+"""
+My task:
+    Move the scintillator in and out
+    Change transmission percentage
+    Take OAV images
+    Open and close fast shutter
+"""
+
+
+# def oav_automation_test()
+def feedback_check(
+    device: XBPMFeedback = inject("xbpm_feedback"),
+):
+    yield from bps.trigger(device)
+
+
+def set_transmission_percentage(
+    percentage: float,
+    attenuator: BinaryFilterAttenuator = inject("attenuator"),
+) -> MsgGenerator:
+    yield from bps.abs_set(attenuator, percentage / 100)
```

Comment:
Should: As above, be careful with waiting for things to finish. You probably want this to only return when it's done so you should do:
```suggestion
    yield from bps.abs_set(attenuator, percentage / 100, wait=True)
```
or
```suggestion
    yield from bps.mv(attenuator, percentage / 100)
```